src/parsers/timezone.rs

```diff
@@ -43,11 +43,8 @@ pub(crate) fn parse_allowed_timezone_formats(
         match offset {
             UtcOffsetRecordOrZ::Z => return Some(TimeZone::default()),
             UtcOffsetRecordOrZ::Offset(offset) => {
-                let offset = match offset {
```

Comment:
issue: I think this should still be maintained as minute precision, right?

The logic here primarily tracks with [ParseTimeZoneIdentifier](https://tc39.es/proposal-temporal/#sec-parsetimezoneidentifier), which is called by [ParseTemporalTimeZoneString](https://tc39.es/proposal-temporal/#sec-temporal-parsetemporaltimezonestring)